serverlessworkflow/modules/ROOT/pages/cloud/knative-service-discovery.adoc

```diff
@@ -0,0 +1,292 @@
+
+The testing procedure in this document is based on the serverless-workflow-examples/serverless-workflow-custom-function-knative example application. You can access this example application in link:{kogito_sw_examples_url}/serverless-workflow-custom-function-knative[Kogito Examples] GitHub repository.
+
+.Prerequisites
+
+include::common/_prerequisites.adoc[]
+
+[[about-custom-functions-for-knative]]
+== About {context} custom functions for Knative
+
+{product_name} provides an implementation of xref:core/custom-functions-support.adoc[{context} custom function] through the `knative-serving` add-on to invoke Knative services. It allows you to have a static URI, defining a Knative service, which is used to perform HTTP requests. The Knative service defined in the URI is queried in the current Knative cluster and translated to a valid URL.
+
+For instance, given the following deployed Knative service:
+
+[source,bash]
+----
+$ kn service list
+NAME                              URL                                                                      LATEST                                  AGE     CONDITIONS   READY   REASON
+custom-function-knative-service   http://custom-function-knative-service.default.10.109.169.193.sslip.io   custom-function-knative-service-00001   3h16m   3 OK / 3     True
+----
+
+You can declare a {context} custom function using the Knative service name, like the following:
+
+[source,json]
+----
+  "functions": [
+    {
+      "name": "greet",
+      "type": "custom",
+      "operation": "knative:custom-function-knative-service"
+    }
+  ]
+----
+
+The above function will send a `POST` request to the http://custom-function-knative-service.default.10.109.169.193.sslip.io URL.
+
+[NOTE]
+====
+`GET` requests are not yet supported.
+====
+
+[[about-namespaces]]
+=== About namespaces
+
+Note that in the above example you declared only the name of the service you wanted to call, but not a namespace. In this case, {product_name} will look for a Knative service in the same namespace the workflow service is running.
+
+In case you need to call a Knative service in a different namespace, you can declare the function as:
+
+[source,json]
+----
+  "functions": [
+    {
+      "name": "greet",
+      "type": "custom",
+      "operation": "knative:my_different_namespace/custom-function-knative-service"
+    }
+  ]
+----
+
+In the above example, {product_name} will look for the `custom-function-knative-service` in the `my_different_namespace` namespace.
+
+=== About HTTP resource paths
+
+You can specify the HTTP resource path by adding the `path` property to the function's metadata.
+
+For instance, given the `/path/to/my/resource` HTTP resource path, you can define your function as:
```

Comment:
```suggestion
For an instance, given the `/path/to/my/resource` HTTP resource path, you can define your function as follows:
```

Reply:
"For **_an_** instance"? Are you sure?
My intention is to say the same as "For example".